artiq/master/scheduler.py

```diff
@@ -139,8 +139,12 @@ def submit(self, expid, priority, due_date, flush, pipeline_name):
         if "repo_rev" in expid:
             if expid["repo_rev"] is None:
                 expid["repo_rev"] = self.experiment_db.cur_rev
-            wd, repo_msg = self.experiment_db.repo_backend.request_rev(
-                expid["repo_rev"])
+            wd, repo_msg, pinned_rev = self.experiment_db.repo_backend.request_rev(
+                expid["repo_rev"]
+            )
+            # Mutate expid's repo_rev to that returned from request_rev, in case
+            # a branch was passed instead of a hash
+            expid["repo_rev"] = pinned_rev
```

Comment:
Hey Sebastian, the value from l.141 is passed into `request_rev` and used to determine the hash of the commit being pointed to. If it's already a hash (as it would be if l.141 ran) then it's returned unaltered.

Reply:
I think the term is "reference", i.e. a branch or a tag which can be resolved to a revision. I agree that's better, but it'll need more code changes to rename it everywhere. Happy to do so if you think it's worth it.

Reply:
> Instead, it would be easier if you had a file in which you could store that SHA-1 value under a simple name so you could use that simple name rather than the raw SHA-1 value.

> In Git, *these simple names are called “references” or “refs”*

Sounds right - so "master" or "v1.0" are both refs.